packages/cozy-konnector-libs/src/libs/BaseKonnector.spec.js

```diff
@@ -14,6 +18,37 @@ describe('BaseKonnector', () => {
     konn = new BaseKonnector()
     konn.accountId = 'account-id'
     client.data.updateAttributes.mockReset()
+    client.data.find.mockReset()
+  })
+
+  it('waitForTwoFaCode should wait for 2FA code', async () => {
+    client.data.find.mockReturnValue(
+      asyncResolve({ twofa_code: 'expected code' })
+    )
+    client.data.updateAttributes.mockReturnValue(asyncResolve({}))
+    const code = await konn.waitForTwoFaCode({ heartBeat: 100 })
+    expect(code).toEqual('expected code')
+    expect(client.data.find.mock.calls.length).toEqual(1)
+    expect(client.data.updateAttributes.mock.calls.length).toEqual(3)
+    expect(client.data.updateAttributes.mock.calls[0][2].state).toEqual(
+      'TWOFA_NEEDED.EMAIL'
+    )
+    expect(client.data.updateAttributes.mock.calls[1][2].twofa_code).toEqual(
+      null
+    )
+    expect(client.data.updateAttributes.mock.calls[2][2].twofa_code).toEqual(
+      null
+    )
+  })
+
+  it('waitForTwoFaCode should throw on timeout', async () => {
+    client.data.find.mockReturnValue(asyncResolve({ twofa_code: null }))
+    client.data.updateAttributes.mockReturnValue(asyncResolve({}))
+    try {
+      await konn.waitForTwoFaCode({ timeout: Date.now() })
+    } catch (err) {
+      expect(err.message).toEqual('USER_ACTION_NEEDED.TWOFA_EXPIRED')
+    }
```

Comment:
If you want to test the fact that the function really throws an error, you can use [`. toThrow()`](https://jestjs.io/docs/en/expect#tothrowerror) or [`. toThrowErrorMatchingSnapshot()`](https://jestjs.io/docs/en/expect#tothrowerrormatchingsnapshothint)
Jest would fail this tests if it won't throw any error

Reply:
I tried, `toThrow` but I could not manage to have it detect the error...